[DEP0062] `node --debug` and `node --debug-brk` are invalid

<!--
This is a bug report tracker. New features are discussed in the forum: https://support.delta.chat
Please fill out as much of this form as you can (leaving out stuff that is not applicable is ok).
-->
Hello, thanks for making delta-chat. When trying to set-up a local dev environment, I ran into an issue launching electron.

- Operating System (Linux/Mac/Windows/iOS/Android): Linux
- Delta Chat Version: master (v1.20.3)
- Expected behavior: Able to start electron with `npm run start`
- Actual behavior: Electron exits with a deprecation notice and status 9
- Steps to reproduce the problem: 

1. Clone the project
2. `npm ci` to install dependencies
3. `npm run start` to launch electron
4.  See that electron exits with status 9 and a deprecation notice

I can also reproduce by running the electron command manually:

```sh
electron . --debug --log-debug --log-to-console --disable-http-cache --translation-watch
echo $? # 9
```

- Screenshots:
- Logs:

```
electron: [DEP0062]: `node --debug` and `node --debug-brk` are invalid. Please use `node --inspect` and `node --inspect-brk` instead.
```

---

I temporarily solved the issue the issue by replacing the `--debug` flag from the `start` script in package.json with `--inspect`.

Separately I encountered this error, which prevented `npm run watch` from working:

```
> deltachat-desktop@1.20.3 watch:shared /home/jchook/forks/deltachat-desktop
> npm run watch:shared:translations

npm ERR! missing script: build:frontend:source_map_support
npm ERR!
npm ERR! Did you mean one of these?
npm ERR!     build:frontend:source-map-support
npm ERR!     build:frontend:themes
npm ERR!     build:frontend:scss
```

I fixed this by updating the script in package.json...

```diff
-     "watch:frontend": "npm run build:frontend:source_map_support && npm run watch:frontend:ts & npm run watch:frontend:static & npm run watch:frontend:scss",
+     "watch:frontend": "npm run build:frontend:source-map-support && npm run watch:frontend:ts & npm run watch:frontend:static & npm run watch:frontend:scss",
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -27,6 +27,7 @@
 - show user-visible error in fullscreen media view if mimetype isn't set or not supported
 
 ### Fixed
+- rename `--debug` flag to `--devmode` (in order to fix #2315)
 - fix duplicated contacts in search
 
 ## [1.20.3] - 2021-06-30

--- a/docs/CLI_FLAGS.md
+++ b/docs/CLI_FLAGS.md
@@ -7,7 +7,7 @@ Options:
 | `--multiple-instances` | [Possible unstable] allows you do open multiple deltachat instances\* |
 | **Development Options** | |
 | `--translation-watch` | enable auto-reload for `_locales/_untranslated_en.json` |
-| `--debug` | opens electron devtools and activates `--log-debug` & `--log-to-console` |
+| `--devmode` | opens electron devtools and activates `--log-debug` & `--log-to-console` |
 | **Theme** | |
 | `--theme <themeid>` | set a specific theme (see [THEMES.md](./THEMES.md)) |
 | `--theme-watch` | enable auto-reload for the active theme |

--- a/docs/LOGGING.md
+++ b/docs/LOGGING.md
@@ -11,7 +11,8 @@ Debug messages are disabled by default, enable them with the `--log-debug` flag.
 | `--machine-readable-stacktrace` | Enable JSON stack trace                       |
 | `--no-color`                    | Disable colors in the output of main process  |
 
-> as of 1.3.0 `--debug` enables both `--log-debug` and `--log-to-console`
+> as of 1.21.0 `--devmode` enables both `--log-debug` and `--log-to-console` (formerly `--debug`)
+> ~as of 1.3.0 `--debug` enables both `--log-debug` and `--log-to-console`~
 
 #### Log locations
 

--- a/package.json
+++ b/package.json
@@ -16,7 +16,7 @@
     "lint": "eslint . --ext .js,.jsx,.ts,.tsx",
     "lint:fix-format": "eslint . --ext .js,.jsx,.ts,.tsx --fix || true && npm run fix-formatting",
     "dev": "npm run build && npm run start --",
-    "start": "electron . --debug --log-debug --log-to-console --disable-http-cache --translation-watch",
+    "start": "electron . --devmode --disable-http-cache --translation-watch",
     "clean": "rm -rf ./tsc-dist ./html-dist ./dist",
     "build": "npm run build:shared && npm run build:backend && npm run build:frontend",
     "build:shared": "npm run build:shared:translations && npm run build:shared:version",

--- a/src/main/index.ts
+++ b/src/main/index.ts
@@ -106,7 +106,7 @@ function onReady([logins, _appReady, loadedState]: [
   mainWindow.init(app, { hidden: app.rc['minimized'] })
   initMenu(logHandler)
 
-  if (rc.debug) {
+  if (rc.devmode) {
     devTools.tryInstallReactDevTools()
     mainWindow.toggleDevTools()
   }

--- a/src/main/rc.ts
+++ b/src/main/rc.ts
@@ -7,7 +7,7 @@ const defaults: RC_Config = {
   'machine-readable-stacktrace': false,
   'multiple-instances': false,
   theme: undefined,
-  debug: false,
+  devmode: false,
   'translation-watch': false,
   'theme-watch': false,
   minimized: false,
@@ -15,7 +15,7 @@ const defaults: RC_Config = {
 
 const config = rc('DeltaChat', defaults) as RC_Config
 
-if (config.debug) {
+if (config.devmode) {
   config['log-debug'] = true
   config['log-to-console'] = true
 }

--- a/src/shared/shared-types.d.ts
+++ b/src/shared/shared-types.d.ts
@@ -57,7 +57,7 @@ export interface RC_Config {
   'multiple-instances': boolean
   theme: string | undefined
   'theme-watch': boolean
-  debug: boolean
+  devmode: boolean
   'translation-watch': boolean
   minimized: boolean
 }

